components/filter/filter-dimension-set.js

```diff
@@ -96,6 +126,22 @@ class FilterDimensionSet extends LitElement {
 		this._dispatchDataChangeEvent({ dimensionKey: this.key, changes: new Map([['values', values]]) });
 	}
 
+	_performSearch() {
+		const valueNodes = this._getSlottedNodes();
+
+		valueNodes.forEach(value => {
+			if (this.searchValue === '') {
+				value.hidden = false;
+			} else {
+				if (value.text.toLowerCase().indexOf(this.searchValue.toLowerCase()) > -1) {
+					value.hidden = false;
+				} else {
+					value.hidden = true;
+				}
+			}
```

Comment:
Wondering out loud here, but would it make sense for the hiding/showing  to be done totally by the top-level filter on its list items that it renders, and not worry about reflecting anything down to the dimension-set or dimension-set-value?

Instead of how it's working now where the top-level filter tells the dimension-set a search has happened, which then hides its own values which then gets communicated back up to the top-level filter to hide the rendered list items.

Reply:
Yeah, this was my main "?" when implementing this.  My thinking was - the current implementation is kinda overkill for if they use the `automatic` option, because we could have just kept it all in the filter's internal components.  _But_ if they do `manual`, I have to setup the same system anyway, where I coordinate between the two and keep things in sync.

So I figured it was cleaner (code-wise) to have both go through the same workflow, and had the benefit of keeping the searching logic with the dimension type instead of being another thing filter needed to know (although we can still do that somewhat either way, with a searching function).  If we remove the `search-value` stuff though, _maybe_ that cleans up the logic a bit and it's worth having the two separate ways to take advantage of just modifying the internals when we can.  I already made a decision that I wasn't allowing you to switch search-type halfway through - you should know this from the get-go.

Reply:
I guess it depends on how manual gets implemented. If they just re-render only the values that match the search vs. re-rendering all the values and setting `hidden` on the ones which don't match.

I was under the impression that the only time you'd want manual search would be if there was some kind of paging implemented... in which case returning "all" wouldn't make sense anyway.

Reply:
So doing a _quick_ look at all the current usages of filter, it seems like there are three methods in use:
1. Simple case-insensitive text compare and then hide options
2. Simple case-insensitive text compare and then remove options
3. Call an api to get the results and then remove options (My Courses)

So maybe we could simplify this a lot... automatic search is applied to the local elements and the consumer doesn't care and can't control that at all.  And for automatic, we'll send an event and it's up to the consumer to physically remove the options they want to hide, which triggers a `slotchange` event and re-render of that filter dimension.